esp/tools/soapplus/http.cpp

```diff
@@ -1313,7 +1318,15 @@ int HttpClient::sendStressRequest(StringBuffer& request, HttpStat* stat, Owned<C
                         fprintf(m_logfile, "%s", recvbuf);
                 }
                 else
+                {
+                    if (total_len == 0)
+                    {
```

Comment:
What happened before this change, if len < 0 and total_len == 0 ?  DId it return -1 also ??

Reply:
No it didn't return -1. But I think total_len == 0 means no response from the server so should be treated as an error. What do you think?

Reply:
Is a reply ALWAYS expected ?  If so then yes, if len <= 0 and total_len == 0 I would say its an error.
If a reply is not ALWAYS guaranteed then if len == 0 and total_len == 0 it would be ok.
Also, should the socket be closed in this case ?

Reply:
Yes a reply is always expected. This code is for non-persistent connection, in which case the socket will be automatically closed in the destructor of CSimpleSocket.